WordPress/Classes/ViewRelated/Post/AztecPostViewController.swift

```diff
@@ -150,18 +156,46 @@ class AztecPostViewController: UIViewController {
 
     /// Post being currently edited
     ///
-    fileprivate(set) var post: AbstractPost {
+    dynamic fileprivate(set) var post: AbstractPost {
         didSet {
+            removeObservers(fromPost: oldValue)
+            addObservers(toPost: post)
+
             refreshInterface()
         }
     }
 
 
+    /// Maintainer of state for editor - like for post button
+    ///
+    fileprivate(set) lazy var postEditorStateContext: PostEditorStateContext = {
+        let originalPostStatus: PostStatus
+
+        if let originalPost = self.post.original,
+            let postStatus = originalPost.status,
+            originalPost.hasRemote() {
+            originalPostStatus = PostStatus(rawValue: postStatus) ?? .draft
+        } else {
+            originalPostStatus = .draft
+        }
+
+        // TODO: Determine if user can actually publish to site or not
+        let context = PostEditorStateContext(originalPostStatus: originalPostStatus, userCanPublish: true, delegate: self)
+        let characterCount = self.post.content?.characters.count ?? 0
+
+        context.updated(hasContent: characterCount > 0)
+
+        return context
+    }()
+
     // MARK: - Lifecycle Methods
 
     init(post: AbstractPost) {
         self.post = post
+
         super.init(nibName: nil, bundle: nil)
+
+        addObservers(toPost: post)
```

Comment:
I _think_ it's safe to remove this `addObservers`, since it's already being done in the **didSet**.

Unless i'm missing something!